crates/tabby/src/services/model/mod.rs

```diff
@@ -18,13 +18,13 @@ pub async fn load_code_generation_and_chat(
     Option<PromptInfo>,
     Option<Arc<dyn ChatCompletionStream>>,
 ) {
-    let (engine, prompt_info, chat) = load_completion_and_chat(completion_model, chat_model).await;
-    let code = engine.map(|engine| Arc::new(CodeGeneration::new(engine)));
+    let (engine, prompt_info, chat) = load_completion_and_chat(&completion_model, chat_model).await;
+    let code = engine.map(|engine| Arc::new(CodeGeneration::new(engine, completion_model)));
     (code, prompt_info, chat)
 }
 
 async fn load_completion_and_chat(
-    completion_model: Option<ModelConfig>,
+    completion_model: &Option<ModelConfig>,
```

Comment:
why change this?

Reply:
Because the config passes in will be used later, so this must change to a borrowed args, or it will take over the life time of the config passed in

Reply:
I recommend just clone it in this particular case.